File: week_6/day_2/shopping/customer/customer.js
var store = require( '../items/items');
var basket = require( '../basket/basket')

var customer = {
  name: "Brian",
  card: true,
  remove: function(){
    basket.contents.pop()
  },
  pick: function(){
    basket.contents.push(store[0]);
  },
  addItem: function (item) {
    for (var i = 0; i < store.length; i++) {
      if(item === store[i]['name']) {
        basket.contents.push(store[i])
      }
    }
  },
  removeItem: function(item) {
   for (var i = 0; i < basket.contents.length; i++) {
    if(item === basket.contents[i]['name']) {
      var index = i;
      basket.contents.splice(i, 1);
      
    }
  } 
}

}


module.exports = customer;

File: week_13/code_test/shopping_basket/Customer.java
package shopping_basket;
import shopping_basket.*;
import java.util.*;


public class Customer {

  public String name;
  public Boolean card;
  public Basket basket;
  // public Store store;

  public Customer (String name, Boolean card, Basket basket){
    this.name = name;
    this.card = card;
    this.basket = basket;
  }

  public void addItem (Item item){
    basket.mContents.add(item);
  }

  public void removeItem(Item item){
    for (int i=0; i<this.basket.getSize(); i++){
      if (item.getName() == this.basket.mContents.get(i).getName()){
        basket.mContents.remove(item);
      }
    }
  }

  public String getName(){
    return this.name;
  }

  public Boolean getCard(){
    return this.card;
  }

  public int getBasketSize(){
    return basket.mContents.size();
  }


}
